### ISSUE TITLE
thread 'main' panicked at 'range start index 3 out of range for slice of length 2', library/core/src/slice/index.rs:52:

### ISSUE THREAD

**Issue by Jolvii85** [rhysnewell/rosella#31]
Hi there,

I run rosella with command "rosella recover -i group-MetaGeno_mb2_depth.txt -r MEGAHIT-group-MetaGeno.contigs.fa --output-directory ./rosella_out/ --threads 14"
But after it build the index, I got an error like: 
thread 'main' panicked at 'range start index 3 out of range for slice of length 2', library/core/src/slice/index.rs:52:
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
And it stoped.

Could you help me to check, what's the problem?

Thank you!

**Comment by rhysnewell** [comment]
Hello,

Would you be able to show me what your `group-MetaGeno_mb2_depth.txt` file looks like?

Cheers,
Rhys

**Comment by Jolvii85** [comment]
> Hello,
> 
> Would you be able to show me what your `group-MetaGeno_mb2_depth.txt` file looks like?
> 
> Cheers, Rhys
Hi Rhys, 
The depth file looks like this:
k141_7783846    23.7211
k141_4245736    7.71226
k141_21228664   2.38346
k141_19813420   3.75191
k141_15567689   12

Thank you for your help!

**Comment by Jolvii85** [comment]
Hi Rhys,

I checked the depth file, it seems like the format of my file is wrong. After I feed the correct depth file to rosella, it can run without errors.

Thank you very much!

Best,

**Comment by rhysnewell** [comment]
Hi,

Good job figuring out the depth file. Do you have any indication of the quality of the assembly you are trying to provide to rosella is? What is the N50, how many contigs, and what is the overall assembly size in base pairs?

Cheers,
Rhys

**Comment by Jolvii85** [comment]
Hi Rhys,

I used all the assembly from the MEGAHIT, maybe I need to filter the raw contigs with length (e.g. minimum 1000 bp) first, then feed them to rosella？

# contigs (>= 0 bp) 22503179
# contigs (>= 1000 bp) 3277097
# contigs (>= 5000 bp) 263856
# contigs (>= 10000 bp) 86370
# contigs (>= 25000 bp) 19635
# contigs (>= 50000 bp) 5314
Total length (>= 0 bp) 17640040897
Total length (>= 1000 bp) 8556999382
Total length (>= 5000 bp) 3101574521
Total length (>= 10000 bp) 1903096983
Total length (>= 25000 bp) 918924844
Total length (>= 50000 bp) 434733433
# contigs 10296633
Largest contig 713443
Total length 13288242988
GC (%) 47.28
N50 1527
N75 781
L50 1701970
L75 4882146
# N's per 100 kbp 0.00
# predicted rRNA genes 1421 + 572 part

Thanks for all your help.

Li Zhao

**Comment by rhysnewell** [comment]
Hi Li,

I'm unsure what would be causing this issue. I'll have to take a closer look, would you be able to send me through the coverage file and the assembly and I'll see if I can get a more informative error to be produced?

Cheers,
Rhys

**Comment by Jolvii85** [comment]
Hi Rhys,
The size of the files are big. I have sent the link to your Gmail.

Just in case, I also paste it here:
https://drive.google.com/file/d/1vOnQuDqnRaxasqMGPXInzpXjhrfxfTBo/view?usp=share_link
https://drive.google.com/file/d/1XymHT6NbNJAveejda7RUbl9etC5BFgqR/view?usp=share_link


Thanks for all your help.

Best,

Li

**Comment by rhysnewell** [comment]
Hi Li,

Sorry have just gotten around to this properly. These files are really big, I'm wondering if your machine is running out memory trying to process them? How much RAM does your machine have?

cheers,
Rhys

**Comment by Jolvii85** [comment]
Hi Rhys,
The node I used has 3.72T RAM, so it should not be a memory error. 

Best,

Li

**Comment by rhysnewell** [comment]
Hi Li,

Would you be able to check something for me?
Could you run flight by itself? i.e.
```
flight bin --input MEGAHIT-group-MetaGeno-depth.txt --kmer_frequencies rosella_test/rosella_kmer_table.tsv --assembly MEGAHIT-group-MetaGeno.contigs.fa --cores 8 --output_directory rosella_test/
```

replacing the kmer table with the one previously produced and the output folder with the previous one as well? And then post what the error says? I think there might be a strange dependency issue occuring with skbio

Cheers,
Rhys

**Comment by rhysnewell** [comment]
I'm hoping this is fixed in `v0.5.0`, error should at least be exposed. Please create a new issue with the error from flight if you still experience this
